src/topology/nhds_set.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2022 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn, Patrick Massot
+-/
+import topology.basic
+/-!
+# Neighborhoods of a set
+
+In this file we define the filter `𝓝ˢ s` or `nhds_set s` consisting of all neighborhoods of a set
+`s`.
+-/
+
+open set filter
+open_locale topological_space
+
+variables {α : Type*} [topological_space α] {s t s₁ s₂ t₁ t₂ : set α} {x : α}
+
+/-- The filter of neighborhoods of a set in a topological space. -/
+def nhds_set (s : set α) : filter α :=
+Sup (nhds '' s)
+
+localized "notation `𝓝ˢ` := nhds_set" in topological_space
+
+lemma mem_nhds_set_iff : t ∈ 𝓝ˢ s ↔ ∀ (x : α), x ∈ s → t ∈ 𝓝 x :=
+by simp_rw [nhds_set, filter.mem_Sup, ball_image_iff]
+
+lemma subset_interior_iff_mem_nhds_set : s ⊆ interior t ↔ t ∈ 𝓝ˢ s :=
+by simp_rw [mem_nhds_set_iff, subset_interior_iff_nhds]
+
+lemma mem_nhds_set : s ∈ 𝓝ˢ t ↔ ∃ U, is_open U ∧ t ⊆ U ∧ U ⊆ s :=
```

Comment:
Minor nitpick but I'm not sure about the naming here.

It seems like the two lemmas currently named `mem_nhds_set_iff`, `mem_nhds_set` might be better named `mem_nhds_set_iff`, `mem_nhds_set_iff'`.

I'll let you decide.

Reply:
You're right!